### PR TITLE
dBFT: properly track chain downloader events

### LINKED ISSUE #265: After the consensus is updated, the local node does not update its own height
#### System information

Geth version: `geth version`
CL client & version: e.g. lighthouse/nimbus/prysm@v1.0.0
OS & Version: Windows/Linux/OSX
Commit hash : (if `develop`)

The new consensus node is deployed only after the consensus switchover, and the new consensus node does not update its height

#### Expected behaviour
The new consensus node is deployed only after the consensus switch, and the new consensus node synchronously updates its height

#### Actual behaviour
The new consensus node is deployed only after the consensus switchover, and the new consensus node does not update its height

#### Steps to reproduce the behaviour
1. The new consensus node is deployed only after the consensus node is switched to a new consensus node
Result This new consensus node does not update its own block

#### Backtrace

````
[backtrace]
````
INFO [07-16|03:40:52.408] Can't validate payload                   hash=228c4b..f429d3 err="message sender is not a validator: expected 0x132d9e2f7Be03542cf09c932f33Eaff46F27cefc, got 0x0177E0C2eF60DcBac10d6EECDd5C7fd84178058b"

2024-07-16T03:56:12.434Z        DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "Commit", "from": 3, "height": 601, "view": 1, "my_height": 1, "my_view": 0}

image

image

[node.log](https://github.com/user-attachments/files/16245408/node.log)

### PR DESCRIPTION
**Pull request by AnnaShaleva**
dBFT should be aware of chain downloader operations since it may stop its work once miner is suspended and there's no trigger to resume dBFT's work once miner operation is resumed. Note that, following miner behaviour, downloader events subscription is one-off and aimed to track only the initial downloader attempt to sync.

Close #265.

Newly-joined CN logs before the bug is fixed:
```
INFO [07-16|12:17:01.524] Initializing BFT consensus               account=0x351FbbaE28a159c7Ff2884Bef497d24ADF003DA4
INFO [07-16|12:17:01.524] Snap sync complete, auto disabling
INFO [07-16|12:17:01.524] Fetching latest sealing proposal         "desired number"=1
INFO [07-16|12:17:01.524] Commit new sealing work                  number=1 sealhash=cfea25..206fb0 "parent hash"=209739..d8931f etherbase=0x1212000000000000000000000000000000000003 txs=0 gas=0 fees=0 elapsed="187.034µs"
INFO [07-16|12:17:02.525] Sealing proposal updated                 number=1 sealhash=cfea25..206fb0 "parent hash"=209739..d8931f txs=0
INFO [07-16|12:17:02.525] Starting dBFT engine                     "last height"=0 "last timestamp"=0
2024-07-16T12:17:02.526+0300    INFO    dbft@v0.2.0/dbft.go:107 initializing dbft       {"height": 1, "view": 0, "index": -1, "role": "WatchOnly"}
INFO [07-16|12:17:02.526] dBFT event loop started
2024-07-16T12:17:02.772+0300    DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "PrepareRequest", "from": 3, "height": 39, "view": 0, "my_height": 1, "my_view": 0}
2024-07-16T12:17:02.772+0300    DEBUG   dbft@v0.2.0/dbft.go:229 caching message from future     {"height": 39, "view": 0, "cache": null}
2024-07-16T12:17:02.776+0300    DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "PrepareResponse", "from": 1, "height": 39, "view": 0, "my_height": 1, "my_view": 0}

... <- some messages cached

16|12:17:02.862] Block synchronisation started
INFO [07-16|12:17:02.862] Mining aborted due to sync
INFO [07-16|12:17:02.898] Imported new chain segment               number=7 hash=793639..f6d9e8 blocks=7 txs=0 mgas=0.000 elapsed=25.458ms    mgasps=0.000 age=3m35s    triedirty=0.00B
INFO [07-16|12:17:02.899] Skipping new block since miner is interrupted "block index"=7 "dbft index"=1
INFO [07-16|12:17:02.900] Indexed transactions                     blocks=8 txs=0 tail=0 elapsed=1.357ms
INFO [07-16|12:17:02.971] Imported new chain segment               number=39 hash=7e5a71..b2a725 blocks=32 txs=8 mgas=1.134 elapsed=71.527ms    mgasps=15.847 snapdiffs=3.67KiB triedirty=32.41KiB
INFO [07-16|12:17:02.971] Skipping new block since miner is interrupted "block index"=39 "dbft index"=1
INFO [07-16|12:17:02.972] Commit new sealing work                  number=40 sealhash=ccd2e9..ba93a8 "parent hash"=7e5a71..b2a725 etherbase=0x1212000000000000000000000000000000000003 txs=0 gas=0 fees=0 elapsed="675.61µs"
2024-07-16T12:17:12.781+0300    DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "RecoveryRequest", "from": 3, "height": 40, "view": 0, "my_height": 1, "my_view": 0}
2024-07-16T12:17:12.781+0300    DEBUG   dbft@v0.2.0/dbft.go:229 caching message from future     {"height": 40, "view": 0, "cache": null}
2024-07-16T12:17:12.782+0300    DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "RecoveryRequest", "from": 2, "height": 40, "view": 0, "my_height": 1, "my_view": 0}
2024-07-16T12:17:12.782+0300    DEBUG   dbft@v0.2.0/dbft.go:229 caching message from future     {"height": 40, "view": 0, "cache": {}}
2024-07-16T12:17:12.783+0300    DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "RecoveryRequest", "from": 1, "height": 40, "view": 0, "my_height": 1, "my_view": 0}
2024-07-16T12:17:12.783+0300    DEBUG   dbft@v0.2.0/dbft.go:229 caching message from future     {"height": 40, "view": 0, "cache": {}}
2024-07-16T12:17:12.783+0300    DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "RecoveryRequest", "from": 3, "height": 40, "view": 0, "my_height": 1, "my_view": 0}
2024-07-16T12:17:12.783+0300    DEBUG   dbft@v0.2.0/dbft.go:229 caching message from future     {"height": 40, "view": 0, "cache": {}}

... <- here goes endless message caching in case of the number of new CNs > M, dBFT never updates its height because miner operation is interrupted

```

And here are logs of the same node after the bug is fixed, so fresh miner proposal is fetched immediately after chain sync ends:

```
INFO [07-16|13:42:13.296] Initializing BFT consensus               account=0x351FbbaE28a159c7Ff2884Bef497d24ADF003DA4
INFO [07-16|13:42:13.296] Snap sync complete, auto disabling
INFO [07-16|13:42:13.296] Fetching latest sealing proposal         "desired number"=1
INFO [07-16|13:42:13.296] Commit new sealing work                  number=1 sealhash=5659e8..8d8c66 "parent hash"=209739..d8931f etherbase=0x1212000000000000000000000000000000000003 txs=0 gas=0 fees=0 elapsed="189.63µs"
INFO [07-16|13:42:14.297] Sealing proposal updated                 number=1 sealhash=5659e8..8d8c66 "parent hash"=209739..d8931f txs=0
INFO [07-16|13:42:14.297] Starting dBFT engine                     "last height"=0 "last timestamp"=0
2024-07-16T13:42:14.298+0300    INFO    dbft@v0.2.0/dbft.go:107 initializing dbft       {"height": 1, "view": 0, "index": -1, "role": "WatchOnly"}
INFO [07-16|13:42:14.298] dBFT event loop started
INFO [07-16|13:42:14.636] Block synchronisation started
INFO [07-16|13:42:14.637] Mining aborted due to sync
INFO [07-16|13:42:14.692] Imported new chain segment               number=9 hash=423246..d41614 blocks=9 txs=0 mgas=0.000 elapsed=43.097ms   mgasps=0.000 age=2m53s     triedirty=0.00B
INFO [07-16|13:42:14.692] Skipping dBFT block callback due to sync "block index"=9 "dbft index"=1
INFO [07-16|13:42:14.694] Indexed transactions                     blocks=10 txs=0 tail=0 elapsed=1.561ms
INFO [07-16|13:42:14.770] Imported new chain segment               number=35 hash=3302d8..4843d5 blocks=26 txs=8 mgas=1.134 elapsed=77.837ms   mgasps=14.563 snapdiffs=3.62KiB triedirty=31.95KiB
INFO [07-16|13:42:14.771] Skipping dBFT block callback due to sync "block index"=35 "dbft index"=1
INFO [07-16|13:42:14.771] New block in the chain                   "dbft index"=1 "chain index"=35 hash=0x3302d8507f819adfda0d785be5777efd95caa311dd3eaa834b255b373c4843d5 "parent hash"=0xa69d5ddd92f7b55d7f5741ff2d7f26efd7bb13276dbeaa376c9809cfd5f0c6d9 primary=3 coinbase=0x1212000000000000000000000000000000000003 "mix digest"=0xf56df5fa6d4dbb0c9eb1a93aa4cf9a4d3a869d142fd0d632231c9a269c7d4f12
INFO [07-16|13:42:14.771] Fetching latest sealing proposal         "desired number"=36
INFO [07-16|13:42:14.773] Commit new sealing work                  number=36 sealhash=1a9515..c2aa06 "parent hash"=3302d8..4843d5 etherbase=0x1212000000000000000000000000000000000003 txs=0 gas=0 fees=0 elapsed=1.988ms
INFO [07-16|13:42:14.773] Commit new sealing work                  number=36 sealhash=1a9515..c2aa06 "parent hash"=3302d8..4843d5 etherbase=0x1212000000000000000000000000000000000003 txs=0 gas=0 fees=0 elapsed="385.899µs"
INFO [07-16|13:42:15.771] Sealing proposal updated                 number=36 sealhash=1a9515..c2aa06 "parent hash"=3302d8..4843d5 txs=0
2024-07-16T13:42:15.772+0300    INFO    dbft@v0.2.0/dbft.go:107 initializing dbft       {"height": 36, "view": 0, "index": 0, "role": "Primary"}
2024-07-16T13:42:15.772+0300    DEBUG   dbft@v0.2.0/dbft.go:608 reset timer     {"h": 36, "v": 0, "delay": "5s"}
2024-07-16T13:42:20.774+0300    DEBUG   dbft@v0.2.0/dbft.go:185 timeout {"height": 36, "view": 0}
2024-07-16T13:42:20.775+0300    DEBUG   dbft@v0.2.0/send.go:8   broadcasting message    {"type": "PrepareRequest", "height": 36, "view": 0}
2024-07-16T13:42:20.776+0300    INFO    dbft@v0.2.0/send.go:35  sending PrepareRequest  {"height": 36, "view": 0}
2024-07-16T13:42:20.776+0300    DEBUG   dbft@v0.2.0/dbft.go:608 reset timer     {"h": 36, "v": 0, "delay": "5s"}
2024-07-16T13:42:20.776+0300    DEBUG   dbft@v0.2.0/check.go:28 check preparations      {"hasReq": true, "count": 1, "M": 3}
```

@txhsl, please review. @songb2, @moonpyt I tested locally with one new CN, it properly updates dBFT context once the node is synced with the network; you may test it with your scenario when all CNs are replaced by the new ones.

### CODE REVIEW

File: consensus/dbft/dbft.go [hunk 1]
```diff
@@ -1294,6 +1313,43 @@ events:
...
+						"index", latest.Number.Uint64(),
+						"err", err.Error())
+					break events
+				}
+
+			case downloader.DoneEvent:
+				c.syncing.Store(false)
+
+				// Stop reacting to downloader events.
+				downloaderEvents.Unsubscribe()
```
Comment (txhsl) [inline consensus/dbft/dbft.go:1342]: If unsubscribe here, will the event loop listen again when the downloader start working next time?

Comment (AnnaShaleva) [reply consensus/dbft/dbft.go:1342]: It won’t, and that’s the idea. This subscription is for a single usage only, exactly like miner does, please, see the method comment.

**Review by txhsl — APPROVED** [review]

**Comment by AnnaShaleva** [comment]
Let’s wait for the testing results for all-consensus-nodes-are-new case before the merge, or I’ll test it by myself in a couple of hours and write back.

**Comment by moonpyt** [comment]
I've verified it on devnet and it's great  @AnnaShaleva 
```
INFO [07-17|06:43:43.789] Maximum peer count                       ETH=30 total=30
INFO [07-17|06:43:43.791] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [07-17|06:43:43.792] Enabling recording of key preimages since archive mode is used
WARN [07-17|06:43:43.792] Disabled transaction unindexing for archive node
INFO [07-17|06:43:43.793] Set global gas cap                       cap=50,000,000
INFO [07-17|06:43:43.793] Initializing the KZG library             backend=gokzg
INFO [07-17|06:43:43.829] Allocated trie memory caches             clean=307.00MiB dirty=0.00B
INFO [07-17|06:43:43.829] Using pebble as the backing database
INFO [07-17|06:43:43.829] Allocated cache and file handles         database=/data/nodes/node7/geth/chaindata cache=512.00MiB handles=524,288
INFO [07-17|06:43:43.852] Opened ancient database                  database=/data/nodes/node7/geth/chaindata/ancient/chain readonly=false
INFO [07-17|06:43:43.852] State scheme set to already existing     scheme=hash
INFO [07-17|06:43:43.853] Initialising Ethereum protocol           network=2,407,101,928 dbversion=<nil>
INFO [07-17|06:43:43.856] 
INFO [07-17|06:43:43.856] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [07-17|06:43:43.856] Chain ID:  2407101928 (unknown)
INFO [07-17|06:43:43.856] Consensus: dBFT (proof-of-authority)
INFO [07-17|06:43:43.856] 
INFO [07-17|06:43:43.856] Pre-Merge hard forks (block based):
INFO [07-17|06:43:43.856]  - Homestead:                   #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/homestead.md)
INFO [07-17|06:43:43.856]  - Tangerine Whistle (EIP 150): #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/tangerine-whistle.md)
INFO [07-17|06:43:43.856]  - Spurious Dragon/1 (EIP 155): #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [07-17|06:43:43.856]  - Spurious Dragon/2 (EIP 158): #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [07-17|06:43:43.856]  - Byzantium:                   #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/byzantium.md)
INFO [07-17|06:43:43.856]  - Constantinople:              #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/constantinople.md)
INFO [07-17|06:43:43.856]  - Petersburg:                  #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/petersburg.md)
INFO [07-17|06:43:43.856]  - Istanbul:                    #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/istanbul.md)
INFO [07-17|06:43:43.856]  - Muir Glacier:                #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/muir-glacier.md)
INFO [07-17|06:43:43.856]  - Berlin:                      #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/berlin.md)
INFO [07-17|06:43:43.856]  - London:                      #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/london.md)
INFO [07-17|06:43:43.856]  - Arrow Glacier:               #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/arrow-glacier.md)
INFO [07-17|06:43:43.856]  - Gray Glacier:                #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/gray-glacier.md)
INFO [07-17|06:43:43.856] 
INFO [07-17|06:43:43.856] The Merge is not yet available for this network!
INFO [07-17|06:43:43.856]  - Hard-fork specification: https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/paris.md
INFO [07-17|06:43:43.856] 
INFO [07-17|06:43:43.856] Post-Merge hard forks (timestamp based):
INFO [07-17|06:43:43.856]  - Shanghai:                    @0          (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/shanghai.md)
INFO [07-17|06:43:43.856]  - NeoXBurn:                      @0         
INFO [07-17|06:43:43.856] 
INFO [07-17|06:43:43.856] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [07-17|06:43:43.856] 
INFO [07-17|06:43:43.856] Loaded most recent local block           number=0 hash=073ed3..5d0ef7 td=1 age=55y4mo1d
WARN [07-17|06:43:43.856] Failed to load snapshot                  err="missing or corrupted snapshot"
INFO [07-17|06:43:43.859] Rebuilding state snapshot
INFO [07-17|06:43:43.861] Initialized transaction indexer          range="entire chain"
INFO [07-17|06:43:43.861] Resuming state snapshot generation       root=44a4f6..768a83 accounts=0 slots=0 storage=0.00B dangling=0 elapsed=1.395ms
INFO [07-17|06:43:43.866] Generated state snapshot                 accounts=25 slots=63 storage=6.80KiB dangling=0 elapsed=6.159ms
INFO [07-17|06:43:43.872] Gasprice oracle is ignoring threshold set threshold=2
WARN [07-17|06:43:43.873] Engine API enabled                       protocol=eth
WARN [07-17|06:43:43.873] Engine API started but chain not configured for merge yet
INFO [07-17|06:43:43.873] Starting peer-to-peer node               instance=Geth/nodes/node7/v0.2.1-unstable-27dfba11-20240716/linux-amd64/go1.22.2
INFO [07-17|06:43:43.883] New local node record                    seq=1,721,198,623,882 id=d03389df4bbdad6a ip=10.128.0.39 udp=30307 tcp=30307
INFO [07-17|06:43:43.883] Started P2P networking                   self=enode://514f46d027eaf8b58453fc5bd35aa8fd9274677feaaf50a793140cff9fa6b4f2ba934d11747d4289ba62f868883180a029acde0a6ec446d47fbdd38485843257@10.128.0.39:30307
INFO [07-17|06:43:43.884] IPC endpoint opened                      url=/data/nodes/node7/geth.ipc
INFO [07-17|06:43:43.884] Generated JWT secret                     path=/data/nodes/node7/geth/jwtsecret
INFO [07-17|06:43:43.885] WebSocket enabled                        url=ws://127.0.0.1:8568
INFO [07-17|06:43:43.885] HTTP server started                      endpoint=127.0.0.1:8568 auth=true prefix= cors=localhost vhosts=localhost
INFO [07-17|06:43:44.596] Unlocked account                         address=0x1c50aAfaB232AAC2D4348d6f5284aB82878a6F79
INFO [07-17|06:43:44.596] Legacy pool tip threshold updated        tip=0
INFO [07-17|06:43:44.596] Legacy pool tip threshold updated        tip=1,000,000,000
INFO [07-17|06:43:44.596] Initializing BFT consensus               account=0x1c50aAfaB232AAC2D4348d6f5284aB82878a6F79
INFO [07-17|06:43:44.596] Fetching latest sealing proposal         "desired number"=1
INFO [07-17|06:43:44.596] Commit new sealing work                  number=1 sealhash=fa4946..080180 "parent hash"=073ed3..5d0ef7 etherbase=0x1212000000000000000000000000000000000003 txs=0 gas=0 fees=0 elapsed="239.65µs"
INFO [07-17|06:43:45.597] Sealing proposal updated                 number=1 sealhash=fa4946..080180 "parent hash"=073ed3..5d0ef7 txs=0
INFO [07-17|06:43:45.597] Starting dBFT engine                     "last height"=0 "last timestamp"=0
2024-07-17T06:43:45.597Z        INFO    dbft@v0.2.0/dbft.go:107 initializing dbft       {"height": 1, "view": 0, "index": -1, "role": "WatchOnly"}
INFO [07-17|06:43:45.597] dBFT event loop started
INFO [07-17|06:43:50.791] Block synchronisation started
INFO [07-17|06:43:50.792] Mining aborted due to sync
INFO [07-17|06:43:50.906] Imported new chain segment               number=33 hash=ae1dba..1af8d1 blocks=33 txs=0 mgas=0.000 elapsed=101.998ms  mgasps=0.000 age=1d21h36m snapdiffs=167.00B triedirty=0.00B
INFO [07-17|06:43:50.906] Skipping dBFT block callback due to sync "block index"=33 "dbft index"=1
INFO [07-17|06:43:50.908] Indexed transactions                     blocks=34 txs=0 tail=0 elapsed=1.524ms
INFO [07-17|06:43:57.644] Imported new chain segment               number=2081 hash=2f94b4..422b3c blocks=2048 txs=50 mgas=9.928 elapsed=6.725s     mgasps=1.476 age=22h40m14s snapdiffs=63.13KiB triedirty=0.00B
INFO [07-17|06:43:57.644] Skipping dBFT block callback due to sync "block index"=2081 "dbft index"=1
INFO [07-17|06:44:03.840] Imported new chain segment               number=4129 hash=db143b..1b8994 blocks=2048 txs=21 mgas=1.233 elapsed=6.186s     mgasps=0.199 age=16h57m13s snapdiffs=112.56KiB triedirty=0.00B
INFO [07-17|06:44:03.840] Skipping dBFT block callback due to sync "block index"=4129 "dbft index"=1
INFO [07-17|06:44:04.709] Can't validate payload                   hash=585a7d..ee366e err="message sender is not a validator: expected 0xFa50A97BECD346b242B661891d6B9B6e1073830E, got 0x9D3FB5C3985f2DcEfC9F55822623A6AEa78F84CD"
INFO [07-17|06:44:04.709] Can't validate payload                   hash=585a7d..ee366e err="message sender is not a validator: expected 0xFa50A97BECD346b242B661891d6B9B6e1073830E, got 0x9D3FB5C3985f2DcEfC9F55822623A6AEa78F84CD"
INFO [07-17|06:44:04.709] Can't validate payload                   hash=585a7d..ee366e err="message sender is not a validator: expected 0xFa50A97BECD346b242B661891d6B9B6e1073830E, got 0x9D3FB5C3985f2DcEfC9F55822623A6AEa78F84CD"
INFO [07-17|06:44:04.710] Can't validate payload                   hash=585a7d..ee366e err="message sender is not a validator: expected 0xFa50A97BECD346b242B661891d6B9B6e1073830E, got 0x9D3FB5C3985f2DcEfC9F55822623A6AEa78F84CD"
INFO [07-17|06:44:04.710] Can't validate payload                   hash=585a7d..ee366e err="message sender is not a validator: expected 0xFa50A97BECD346b242B661891d6B9B6e1073830E, got 0x9D3FB5C3985f2DcEfC9F55822623A6AEa78F84CD"
INFO [07-17|06:44:04.710] Can't validate payload                   hash=585a7d..ee366e err="message sender is not a validator: expected 0xFa50A97BECD346b242B661891d6B9B6e1073830E, got 0x9D3FB5C3985f2DcEfC9F55822623A6AEa78F84CD"
INFO [07-17|06:44:07.805] Can't validate payload                   hash=485fb9..ea8e1e err="message sender is not a validator: expected 0xe30c5f0dD77B8ED0d79cA5714820714A42acea88, got 0x3F4C66F8146d9D4D13172e0FB4438617A10a7d08"
INFO [07-17|06:44:07.805] Can't validate payload                   hash=485fb9..ea8e1e err="message sender is not a validator: expected 0xe30c5f0dD77B8ED0d79cA5714820714A42acea88, got 0x3F4C66F8146d9D4D13172e0FB4438617A10a7d08"
INFO [07-17|06:44:07.806] Can't validate payload                   hash=485fb9..ea8e1e err="message sender is not a validator: expected 0xe30c5f0dD77B8ED0d79cA5714820714A42acea88, got 0x3F4C66F8146d9D4D13172e0FB4438617A10a7d08"
INFO [07-17|06:44:09.934] Imported new chain segment               number=6177 hash=ea7ba7..026457 blocks=2048 txs=0  mgas=0.000 elapsed=6.083s     mgasps=0.000 age=11h14m8s  snapdiffs=142.11KiB triedirty=0.00B
INFO [07-17|06:44:09.934] Skipping dBFT block callback due to sync "block index"=6177 "dbft index"=1
INFO [07-17|06:44:10.757] Can't validate payload                   hash=71dae4..0be898 err="message sender is not a validator: expected 0xdf1f3BAE73962a1ED67eE7BE35DDeaD361cd3232, got 0x30c4A56f9b1D72bea20585DFC5a1D04c4d90A5B9"
INFO [07-17|06:44:10.757] Can't validate payload                   hash=71dae4..0be898 err="message sender is not a validator: expected 0xdf1f3BAE73962a1ED67eE7BE35DDeaD361cd3232, got 0x30c4A56f9b1D72bea20585DFC5a1D04c4d90A5B9"
INFO [07-17|06:44:13.191] Can't validate payload                   hash=f396cd..99f8b1 err="message sender is not a validator: expected 0x132d9e2f7Be03542cf09c932f33Eaff46F27cefc, got 0x0177E0C2eF60DcBac10d6EECDd5C7fd84178058b"
INFO [07-17|06:44:13.191] Can't validate payload                   hash=826510..00a8d9 err="message sender is not a validator: expected 0x67C4c285D4d5b2D479Ac559B4608821F558c972a, got 0x2D0603729A96BDDC0baa2B8AEF0a4058e27C50d4"
INFO [07-17|06:44:13.192] Can't validate payload                   hash=826510..00a8d9 err="message sender is not a validator: expected 0x67C4c285D4d5b2D479Ac559B4608821F558c972a, got 0x2D0603729A96BDDC0baa2B8AEF0a4058e27C50d4"
INFO [07-17|06:44:13.192] Can't validate payload                   hash=826510..00a8d9 err="message sender is not a validator: expected 0x67C4c285D4d5b2D479Ac559B4608821F558c972a, got 0x2D0603729A96BDDC0baa2B8AEF0a4058e27C50d4"
INFO [07-17|06:44:13.192] Can't validate payload                   hash=826510..00a8d9 err="message sender is not a validator: expected 0x67C4c285D4d5b2D479Ac559B4608821F558c972a, got 0x2D0603729A96BDDC0baa2B8AEF0a4058e27C50d4"
INFO [07-17|06:44:15.254] Imported new chain segment               number=8225 hash=7aca17..e01710 blocks=2048 txs=0  mgas=0.000 elapsed=5.311s     mgasps=0.000 age=5h31m9s   snapdiffs=172.10KiB triedirty=0.00B
INFO [07-17|06:44:15.254] Skipping dBFT block callback due to sync "block index"=8225 "dbft index"=1
INFO [07-17|06:44:18.734] Can't validate payload                   hash=8ce5d4..89860c err="message sender is not a validator: expected 0x67C4c285D4d5b2D479Ac559B4608821F558c972a, got 0x2D0603729A96BDDC0baa2B8AEF0a4058e27C50d4"
INFO [07-17|06:44:18.734] Can't validate payload                   hash=8ce5d4..89860c err="message sender is not a validator: expected 0x67C4c285D4d5b2D479Ac559B4608821F558c972a, got 0x2D0603729A96BDDC0baa2B8AEF0a4058e27C50d4"
INFO [07-17|06:44:19.931] Imported new chain segment               number=10026 hash=8bc810..86a5e2 blocks=1801 txs=1  mgas=0.154 elapsed=4.670s     mgasps=0.033 age=22m14s    snapdiffs=199.63KiB triedirty=0.00B
INFO [07-17|06:44:19.931] Skipping dBFT block callback due to sync "block index"=10026 "dbft index"=1
INFO [07-17|06:44:20.875] Imported new chain segment               number=10028 hash=38d1f4..ac3203 blocks=2    txs=0  mgas=0.000 elapsed=7.300ms    mgasps=0.000 age=21m55s    snapdiffs=199.63KiB triedirty=0.00B
INFO [07-17|06:44:20.875] Skipping dBFT block callback due to sync "block index"=10028 "dbft index"=1
INFO [07-17|06:44:20.875] New block in the chain                   "dbft index"=1 "chain index"=10028 hash=0x38d1f4047be260ef58d467d0f9e36ad6edf7e11b7a5082ea43ef55fff8ac3203 "parent hash"=0x74185ddf8b2877ec1009eae162fccbb4d0504dc90015e156a5b823f02646875b primary=4 coinbase=0x1212000000000000000000000000000000000003 "mix digest"=0x2657300556091bdeb22d8869a0c80ecc082135943d6cdcaff928b58ee457ebb9
INFO [07-17|06:44:20.875] Fetching latest sealing proposal         "desired number"=10029
INFO [07-17|06:44:20.875] Commit new sealing work                  number=10029 sealhash=99cf6e..e6dcd3 "parent hash"=38d1f4..ac3203 etherbase=0x1212000000000000000000000000000000000003 txs=0  gas=0 fees=0 elapsed="299.32µs"
INFO [07-17|06:44:20.875] Sealing proposal updated                 number=10029 sealhash=99cf6e..e6dcd3 "parent hash"=38d1f4..ac3203 txs=0
2024-07-17T06:44:20.876Z        INFO    dbft@v0.2.0/dbft.go:107 initializing dbft       {"height": 10029, "view": 0, "index": 1, "role": "Backup"}
2024-07-17T06:44:20.876Z        DEBUG   dbft@v0.2.0/dbft.go:608 reset timer     {"h": 10029, "v": 0, "delay": "20s"}
2024-07-17T06:44:33.271Z        DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "RecoveryRequest", "from": 6, "height": 10029, "view": 0, "my_height": 10029, "my_view": 0}
2024-07-17T06:44:33.271Z        DEBUG   dbft@v0.2.0/send.go:8   broadcasting message    {"type": "RecoveryMessage", "height": 10029, "view": 0}
2024-07-17T06:44:40.877Z        DEBUG   dbft@v0.2.0/dbft.go:185 timeout {"height": 10029, "view": 0}
2024-07-17T06:44:40.877Z        DEBUG   dbft@v0.2.0/dbft.go:608 reset timer     {"h": 10029, "v": 0, "delay": "40s"}
2024-07-17T06:44:40.877Z        INFO    dbft@v0.2.0/send.go:61  skip change view        {"nc": 0, "nf": 5}
2024-07-17T06:44:40.877Z        DEBUG   dbft@v0.2.0/send.go:8   broadcasting message    {"type": "RecoveryRequest", "height": 10029, "view": 0}
2024-07-17T06:44:44.710Z        DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "ChangeView", "from": 5, "height": 10029, "view": 0, "my_height": 10029, "my_view": 0}
2024-07-17T06:44:44.710Z        INFO    dbft@v0.2.0/dbft.go:473 received ChangeView     {"validator": 5, "reason": "Timeout", "new view": 1}
2024-07-17T06:44:44.710Z        DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "ChangeView", "from": 5, "height": 10029, "view": 0, "my_height": 10029, "my_view": 0}
2024-07-17T06:44:44.710Z        INFO    dbft@v0.2.0/dbft.go:473 received ChangeView     {"validator": 5, "reason": "Timeout", "new view": 1}
2024-07-17T06:44:44.710Z        DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "ChangeView", "from": 5, "height": 10029, "view": 0, "my_height": 10029, "my_view": 0}
2024-07-17T06:44:44.710Z        INFO    dbft@v0.2.0/dbft.go:473 received ChangeView     {"validator": 5, "reason": "Timeout", "new view": 1}
2024-07-17T06:44:44.710Z        DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "ChangeView", "from": 5, "height": 10029, "view": 0, "my_height": 10029, "my_view": 0}
2024-07-17T06:44:44.710Z        INFO    dbft@v0.2.0/dbft.go:473 received ChangeView     {"validator": 5, "reason": "Timeout", "new view": 1}
2024-07-17T06:44:44.710Z        DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "ChangeView", "from": 5, "height": 10029, "view": 0, "my_height": 10029, "my_view": 0}
2024-07-17T06:44:44.710Z        INFO    dbft@v0.2.0/dbft.go:473 received ChangeView     {"validator": 5, "reason": "Timeout", "new view": 1}
2024-07-17T06:44:44.710Z        DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "ChangeView", "from": 5, "height": 10029, "view": 0, "my_height": 10029, "my_view": 0}
2024-07-17T06:44:44.710Z        INFO    dbft@v0.2.0/dbft.go:473 received ChangeView     {"validator": 5, "reason": "Timeout", "new view": 1}
2024-07-17T06:44:47.806Z        DEBUG   dbft@v0.2.0/dbft.go:214 received message        {"type": "ChangeView", "from": 4, "height": 10029, "view": 0, "my_height": 10029, "my_view": 0}
2024-07-17T06:44:47.806Z        INFO    dbft@v0.2.0/dbft.go
```

**Comment by AnnaShaleva** [comment]
@moonpyt, good indeed, works as expected.

**Pull request merged**